mule-extensions-api/src/main/java/org/mule/runtime/extension/api/declaration/type/ExtensionsObjectFieldHandler.java

```diff
@@ -137,6 +143,15 @@ private void processParameterGroup(Field field, ObjectFieldTypeBuilder fieldBuil
     if (field.getAnnotation(ParameterGroup.class) != null) {
       fieldBuilder.with(new FlattenedTypeAnnotation());
     }
+
+    ExclusiveOptionals exclusiveOptionals = field.getType().getAnnotation(ExclusiveOptionals.class);
```

Comment:
this has to be done _only_ if the field is also annotated with ParameterGroup, if it is used as a POJO (no group annotation) then exclusive optionals is not supported